pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -155,7 +134,8 @@ func (n Node) hasCompatibleNodeSelector(nodeSelector map[string]string, it cloud
 
 func (n Node) isPodCompatible(pod *v1.Pod, it cloudprovider.InstanceType) bool {
 	return n.hasCompatibleResources(resources.RequestsForPods(pod), it) &&
-		n.hasCompatibleNodeSelector(pod.Spec.NodeSelector, it)
+		n.hasCompatibleNodeSelector(pod.Spec.NodeSelector, it) &&
+		n.hasCompatibleNodeAffinity(pod.Spec.Affinity, it)
```

Comment:
Aren't these checks already modeled by `Requirements`? Does this disappear when we combine scheduling/binpacking? In the mean time, can you use PodRequirements(pod) to handle the selector and affinity parsing?

Reply:
Yup 😞